modules/programs/i3status-rust.nix

```diff
@@ -0,0 +1,179 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.i3status-rust;
+
+  # This is in nixpkgs now, but not in 20.03
+  # TODO: Take from upstrem (https://github.com/NixOS/nixpkgs/blob/master/pkgs/pkgs-lib/formats.nix) once HM no longer tracks 20.03
+  json = { }: {
+    type = with lib.types;
+      let
+        valueType = nullOr
+          (oneOf [ bool int float str (attrsOf valueType) (listOf valueType) ])
+          // {
+            description = "JSON value";
+          };
+      in valueType;
+
+    generate = name: value:
+      pkgs.runCommandNoCC name {
+        nativeBuildInputs = [ pkgs.jq ];
+        value = builtins.toJSON value;
+        passAsFile = [ "value" ];
+      } ''
+        jq . "$valuePath"> $out
+      '';
+  };
+
+  restartI3 = ''
+    i3Socket=''${XDG_RUNTIME_DIR:-/run/user/$UID}/i3/ipc-socket.*
+    if [ -S $i3Socket ]; then
+      echo "Reloading i3"
+      $DRY_RUN_CMD ${config.xsession.windowManager.i3.package}/bin/i3-msg -s $i3Socket restart 1>/dev/null
+    fi
+  '';
+
+  settingsFormat = toml { };
+
+  # This is in nixpkgs now, but not in 20.03
+  # TODO: Take from upstrem (https://github.com/NixOS/nixpkgs/blob/master/pkgs/pkgs-lib/formats.nix) once HM no longer tracks 20.03
+  toml = { }:
+    json { } // {
+      type = with lib.types;
+        let
+          valueType =
+            oneOf [ bool int float str (attrsOf valueType) (listOf valueType) ]
+            // {
+              description = "TOML value";
+            };
+        in valueType;
+
+      generate = name: value:
+        pkgs.runCommandNoCC name {
+          nativeBuildInputs = [ pkgs.remarshal ];
+          value = builtins.toJSON value;
+          passAsFile = [ "value" ];
+        } ''
+          json2toml "$valuePath" "$out"
+        '';
+
+    };
+
+in {
+  meta.maintainers = [ maintainers.farlion ];
+
+  options.programs.i3status-rust = {
+    enable = mkEnableOption "a replacement for i3-status written in Rust";
+
+    blocks = mkOption {
+      type = settingsFormat.type;
+      default = [
+        {
+          block = "disk_space";
+          path = "/";
+          alias = "/";
+          info_type = "available";
+          unit = "GB";
+          interval = 60;
+          warning = 20.0;
+          alert = 10.0;
+        }
+        {
+          block = "memory";
+          display_type = "memory";
+          format_mem = "{Mup}%";
+          format_swap = "{SUp}%";
+        }
+        {
+          block = "cpu";
+          interval = 1;
+        }
+        {
+          block = "load";
+          interval = 1;
+          format = "{1m}";
+        }
+        { block = "sound"; }
+        {
+          block = "time";
+          interval = 60;
+          format = "%a %d/%m %R";
+        }
+      ];
+      description = ''
+        Configuration blocks to add to i3status-rust <filename>config</filename>
+        See
+        https://github.com/greshake/i3status-rust/blob/master/blocks.md 
```

Comment:
Links should be made clickable.

Reply:
:+1: done in 90dd17c9076737ed3e145414a9e239e3f4bbc4cd